### PR TITLE
refactor: run networkd as a goroutine in machined

### PR DESCRIPTION
**Pull request by smira**
This removes machined as a separate container and image.

Reasons:

* `machined` becomes more and more bound into the core flow - now it
interacts with `etcd` for VIPs, so container has more and more
mounts/permissions
* it should be easier to COSIfy machined piece by piece if we have it
running in the same process
* initramfs size

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by Unix4ever — DISMISSED** [review]

**Comment by AlekSi** [comment]
> This removes machined as a separate container and image.

`networkd`, not `machined`, right?

File: internal/app/networkd/main.go [hunk 1]
```diff
@@ -2,66 +2,37 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-package main
+package networkd
```
Comment (AlekSi) [inline internal/app/networkd/main.go:5]: Should we rename this file now that it does not contain the "normal" `main()` function?

Should we also move this file away from `internal/app/` according to the project layout document?

Comment (AlekSi) [reply internal/app/networkd/main.go:5]: Should we also update components documentation?

**Comment by smira** [comment]
> > This removes machined as a separate container and image.
> 
> `networkd`, not `machined`, right?

Yes, of course :facepalm:

File: internal/app/networkd/main.go (hunk 1 above)
Comment (smira) [reply internal/app/networkd/main.go:5]: > Should we rename this file now that it does not contain the "normal" `main()` function?

I would rather keep it in `main.go` for now, as it's still an entrypoint to the service.

Comment (smira) [reply internal/app/networkd/main.go:5]: > Should we also update components documentation?

From the user point of view, nothing has changed, as still service `networkd` exists, it's visible in `talosctl services`, `talosctl logs networkd` works, etc. It just changes the way we run it, but it should be transparent to the user.

Comment (smira) [reply internal/app/networkd/main.go:5]: > Should we also move this file away from `internal/app/` according to the project layout document?

same answer, nothing has changed so far, it's still a service and kind of same service. 

networkd itself is supposed to be removed any time soon once we make it COSIfied, and then this directory will go away and docs will be updated

**Pull request merged**